In [138]:
from scipy.stats import norm
from scipy.stats import expon
from scipy.stats import lognorm
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sb
import math

# 1. The length of an injected-molded plastic case that holds tape is normally distributed with mean length 90.2 mm and a standard deviation of 0.1 mm.

##  a. What is the probability that a part is shorter than 90.3 mm and longer than 89.7 mm?

In [139]:
round(norm(loc = 90.2, scale = 0.1).cdf(90.3) - norm(loc = 90.2, scale = 0.1).cdf(89.7),4)

0.8413

## b. What should the process mean be to obtain the greatest number of parts between 89.7 and 90.3 mm?

In [140]:
m = np.mean([89.7,90.3]) 
print(m)

90.0


## c. If parts that are not between 89.7 and 90.3 mm are scrapped, what is the yield of the process mean that you chose in part b

In [141]:
round(norm(loc = 90.0, scale = 0.1).cdf(90.3) - norm(loc = 90.0, scale = 0.1).cdf(89.7),4)

0.9973

# 2. Extensive experience states that the time to failure of fans in certain types of diesel engines follow an exponential distribution with mean time to failure 25,000 hours.

## a. What is the probability that the fan will last at least 20,000 hours

In [142]:
1 -expon.cdf(20000,scale = 25000)

0.44932896411722156

## b. What is the standard deviation of time to failure?

In [143]:
expon(scale = 25000).std()

25000.0

## c. What is the probability that a fan will last longer than the mean by more than 2 standard deviations?

In [144]:
1 - expon.cdf(75000,scale = 25000)

0.04978706836786395

# A person can be in an upright position either sitting or standing. For children between the ages of eight and fifteen, time per day in an upright position follows a normal probability model with mean 5.4 hours and standard deviation 1.3 hours.

## a. What is the probability that a randomly selected child’s time per day in an uprightposition is more than 6 hours?

In [145]:
round(1 - norm(loc = 5.2, scale = 1.3).cdf(6),4)

0.2692

## b. What is the probability that a randomly selected child’s time per day in an upright position is less than 3 hours?


In [146]:
round(norm(loc = 5.2, scale = 1.3).cdf(3),4)

0.0453

## c. Fill in the blank: 5% of children have less than ____ hours per day in an upright position.



In [147]:
import scipy.stats
my_dist = norm(loc = 5.2, scale = 1.3)
round(my_dist.ppf(0.05),4)

3.0617

## d. Children in the highest 1% for time spent upright were in this position the longest. Therefore, 1% of children have more than ___ hours per day in an upright position.

In [148]:
round(my_dist.ppf(0.99),4)

8.2243

# 4. The life of an Infinity brand car battery is normally distributed with mean 900 days and standard deviation 35 days.


## a. What is the probability that a randomly selected Infinity car battery has a lifetime between 825 and 900 days?



In [149]:
my_dist = norm(loc = 900, scale = 35)
round(my_dist.cdf(900) - my_dist.cdf(825),4)

0.4839

## b. What is the probability that a randomly selected Infinity car battery has a lifetime more than 950 days?

In [150]:
1 - my_dist.cdf(950)

0.07656372550983481

## c. What is the 67th percentile of brand Infinity lifetimes?


In [151]:
round(my_dist.ppf(0.67),4)

915.397

## d. One Infinity battery had a lifetime of 825 days. Is this unusual? EXPLAIN.

In [152]:
round(my_dist.cdf(825),4)


0.0161

No, it is not unusual, as the observation is not less than the mean by more than 3standard deviations, even though it is in lower upper bound.


In [153]:
900-3*35

795

if the variable was less than 795, it would be a outlier.

# 5. Suppose that 𝑋 has a lognormal distribution with parameters 𝜇 = 5 and 𝜎 = 0.3. Determine the following:

## a. 𝑃(𝑋 < 300)

In [156]:
my_log_dist = lognorm(0.3,scale=math.exp(5))
round(my_log_dist.cdf(300),4)

0.9905

## b. The value for 𝑥 such that 𝑃(𝑋 ≤ 𝑥) = 0.95

In [157]:
my_log_dist.ppf(0.95)

243.09591759845222

In [121]:
dist

0.8192571931741357

## c. 𝑃(100 < 𝑋 < 500)

In [158]:
round(my_log_dist.cdf(500)-my_log_dist.cdf(100),4)

0.9059